mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,874 @@
+        spark_conf["spark.python.worker.reuse"] = "true"
+        # disable task retry (i.e. make it fast fail)
+        spark_conf["spark.task.maxFailures"] = "1"
+        # Disable simplifying traceback from Python UDFs
+        spark_conf["spark.sql.execution.pyspark.udf.simplifiedTraceback.enabled"] = "false"
+        # Show jvm side stack trace.
+        spark_conf["spark.sql.pyspark.jvmStacktrace.enabled"] = "true"
+        spark_conf["spark.jars.excludes"] = "net.sourceforge.f2j:arpack_combined_all"
+        spark_conf["spark.driver.extraJavaOptions"] = "-Dio.netty.tryReflectionSetAccessible=true"
+        spark_conf["spark.executor.extraJavaOptions"] = "-Dio.netty.tryReflectionSetAccessible=true"
+
+        os.environ['PYSPARK_PYTHON'] = sys.executable
+        os.environ['PYSPARK_DRIVER_PYTHON'] = sys.executable
+        if model_path:
+            jar_paths, license_path = fetch_deps_from_path(os.path.join(model_path))
+            # jar_paths += get_mleap_jars().split(',')  # TODO when to load MLleap Jars
+            if license_path:
+                if 'JSL_NLP_LICENSE' not in os.environ:
+                    os.environ.update({k: str(v) for k, v in json.load(open(license_path)).items() if v is not None})
+                    os.environ['JSL_NLP_LICENSE'] = json.load(open(license_path))['HC_LICENSE']
+            print("STARTING NEW  SESSION WITH JARS: ", jar_paths)
+            spark = nlp.start(nlp=False, spark_nlp=False, jar_paths=jar_paths, json_license_path=license_path,
+                              create_jsl_home_if_missing=False, spark_conf=spark_conf)
+        else:
+            spark = nlp.start()
+    return spark
+
+
+def create_sparksession_for_jsl_uri(**kwargs):
+    # if using johnsnowlabs:/ uri, we need to start a sparksession
+    # Support johnsnowlabs nlp.start() API to fetch a session
+    nlp.start(**kwargs)
+
+
+def fetch_deps_from_path(local_model_path):
+    if _JOHNSNOWLABS_MODEL_PATH_SUB not in local_model_path:
+        local_model_path = os.path.join(local_model_path, _JOHNSNOWLABS_MODEL_PATH_SUB, 'jars.jsl', )
+    else:
+        local_model_path = os.path.join(local_model_path, 'jars.jsl', )
+
+    jar_paths = [os.path.join(local_model_path, file) for file in os.listdir(local_model_path) if '.jar' in file]
+    license_path = [os.path.join(local_model_path, file) for file in os.listdir(local_model_path) if '.json' in file]
+    license_path = license_path[0] if license_path else None
+    return jar_paths, license_path
+
+
+def _unpack_and_save_model(spark_model, dst):
+    from pyspark.ml import PipelineModel
+    if isinstance(spark_model, _PyFuncModelWrapper):
+        spark_model = spark_model.spark_model
+    if isinstance(spark_model, PipelineModel):
+        spark_model.write().overwrite().save(dst)
+    else:
+        # nlu pipe
+        spark_model.predict('Init')
+        try:
+            spark_model.vanilla_transformer_pipe.write().overwrite().save(dst)
+        except:
```

Comment:
please catch the exact exception that you're targeting

Reply:
There is a random exception happening when using mlflowdbfs which I could not drill down yet.
I am not sure if it might happen somewhere else for different reasons with a different exception, so I like to have this general exception to catch it as a fallback.